### ISSUE TITLE
How to deserialise map with typed values

### ISSUE THREAD

**Issue by rauanmayemir** [json4s/json4s#284]
I have a map that looks like this:

``` scala
trait MyType
case class MyType1() extends MyType
case class MyType2() extends MyType
case class MyType3() extends MyType

val mymap: Map[String, MyType] = Map("type1" -> objOfMyType1, "type2" -> objOfMyType2, "type3" -> objOfMyType3)
```

I've seen only separate examples on custom key or field serializers, but how do I extract those field objects depending on what is the key?

**Comment by seratch** [comment]
Take a look at type hinting. https://github.com/json4s/json4s#serializing-polymorphic-lists

**Comment by rauanmaemirov** [comment]
Yep, I ended up doing that.

But originally my question was how to achieve that without type hints, sorry for not pointing that out.
So I tried doing CustomSerializer, parsing map value by its key, but got lost. Here's the gist of my failed attempt. https://gist.github.com/rauanmaemirov/9cbac5e041ac323e2285

**Comment by ncolomer** [comment]
I achieved to implement what is described here using a `CustomSerializer`.
See PR #454

**Comment by seratch** [comment]
Implemented by #454
